### PR TITLE
Fix upgrade dependency org.apache.pulsar:pulsar-client to v3.3.2

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #11053

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/server/springboot/broker/pulsar/domain/PulsarModuleFactory.java [hunk 1]
```diff
@@ -45,7 +47,8 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
         .put("pulsarDockerImage", dockerImages.get("apachepulsar/pulsar").fullName())
         .and()
       .javaDependencies()
-        .addDependency(groupId("org.apache.pulsar"), artifactId("pulsar-client"), versionSlug("pulsar"))
+        .addDependency(groupId("org.apache.pulsar"), artifactId("pulsar-client"), versionSlug(PULSAR))
+        .addDependency(groupId("org.apache.pulsar"), artifactId("pulsar-client-api"), versionSlug(PULSAR))
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/broker/pulsar/domain/PulsarModuleFactory.java:51]: Why is this needed? This is not mentioned in the official documentation: https://pulsar.apache.org/docs/3.3.x/client-libraries-java-setup/

Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/broker/pulsar/domain/PulsarModuleFactory.java:51]: don't find a better solution for the moment

**Comment by pascalgrimaud** [comment]
maybe try to upgrade to v4 now?

**Comment by qmonmert** [comment]
ok I will try
